servicetalk-encoding-netty/src/test/java/io/servicetalk/encoding/netty/NettyCompressionStreamingSerializerTest.java

```diff
@@ -0,0 +1,121 @@
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.servicetalk.encoding.netty;
+
+import io.servicetalk.buffer.api.Buffer;
+import io.servicetalk.buffer.api.BufferAllocator;
+import io.servicetalk.buffer.api.CompositeBuffer;
+import io.servicetalk.concurrent.api.Publisher;
+import io.servicetalk.serializer.api.SerializerDeserializer;
+import io.servicetalk.serializer.api.StreamingSerializerDeserializer;
+
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.EnumSource;
+
+import java.util.concurrent.ThreadLocalRandom;
+
+import static io.servicetalk.buffer.api.ReadOnlyBufferAllocators.DEFAULT_RO_ALLOCATOR;
+import static io.servicetalk.buffer.netty.BufferAllocators.DEFAULT_ALLOCATOR;
+import static io.servicetalk.concurrent.api.Publisher.from;
+import static io.servicetalk.encoding.netty.NettyCompression.deflateDefault;
+import static io.servicetalk.encoding.netty.NettyCompression.deflateDefaultStreaming;
+import static io.servicetalk.encoding.netty.NettyCompression.gzipDefault;
+import static io.servicetalk.encoding.netty.NettyCompression.gzipDefaultStreaming;
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.greaterThanOrEqualTo;
+
+class NettyCompressionStreamingSerializerTest {
+    enum StreamingType {
+        GZIP(gzipDefaultStreaming(), DEFAULT_ALLOCATOR),
+        DEFLATE(deflateDefaultStreaming(), DEFAULT_ALLOCATOR),
+        GZIP_RO(gzipDefaultStreaming(), DEFAULT_RO_ALLOCATOR),
+        DEFLATE_RO(deflateDefaultStreaming(), DEFAULT_RO_ALLOCATOR);
+
+        final StreamingSerializerDeserializer<Buffer> serializer;
+        final BufferAllocator allocator;
+
+        StreamingType(StreamingSerializerDeserializer<Buffer> serializer, BufferAllocator allocator) {
+            this.serializer = serializer;
+            this.allocator = allocator;
+        }
+    }
+
+    enum AggType {
+        GZIP(gzipDefault(), DEFAULT_ALLOCATOR),
+        DEFLATE(deflateDefault(), DEFAULT_ALLOCATOR),
+        GZIP_RO(gzipDefault(), DEFAULT_RO_ALLOCATOR),
+        DEFLATE_RO(deflateDefault(), DEFAULT_RO_ALLOCATOR);
+
+        final SerializerDeserializer<Buffer> serializer;
+        final BufferAllocator allocator;
+
+        AggType(SerializerDeserializer<Buffer> serializer, BufferAllocator allocator) {
+            this.serializer = serializer;
+            this.allocator = allocator;
+        }
+    }
+
+    @ParameterizedTest(name = "{index}: type = {0}")
```

Comment:
```suggestion
    @ParameterizedTest(name = "{displayName} [{index}] type = {0}")
```
For Intellij IDEA `displayName` is not important, but in jUnit HTML report that will be published on CI the name won't be included without it.